state-chain/pallets/cf-flip/src/on_charge_transaction.rs

```diff
@@ -67,6 +88,10 @@ impl<T: TxConfig + FlipConfig + Config> OnChargeTransaction<T> for FlipTransacti
 				surplus.peek()
 			};
 
+			let to_burn =
```

Comment:
This seems very dangerous now. If we have an exponential fee multiplier, then the to_burn can get *very* large very quickly... and this to_burn is passed directly into Flip::burn... meaning we could burn basically everything, just from one person spamming.
1. We should explicitly ceil the to_burn to the pre_dispatch max.
2. We should have a test for this.

Reply:
You can't burn more than is in the account, the Surplus/Deficit types make sure of this. (see imbalances.rs)